.ci/jenkins/build-test/Jenkinsfile

```diff
@@ -21,6 +21,71 @@ pipeline {
                      required: true)
     }
     stages {
+        stage('Scan for possible secrets'){
+            steps{
+                sh label: "Secrets scan",
+                   script: '''
+                   #!/bin/bash
+
+                   set +x
+
+                    if [ ! -f .secrets.baseline ] ;
+                    then
+                        # This generated baseline file will only be temporarily available on the GitHub side and will not appear in the user's local files.
+                        # Scanning an empty folder to generate an initial .secrets.baseline without secrets in the results.
+                        echo "⚠️ No existing .secrets.baseline file detected. Creating a new blank baseline file."
+                        mkdir empty-dir
+                        detect-secrets scan empty-dir > .secrets.baseline
+                        echo "✅ Blank .secrets.baseline file created successfully."
+                        rm -r empty-dir
+                    else
+                        echo "✅ Existing .secrets.baseline file detected. No new baseline file will be created."
+                    fi
+                    # scripts scan repository for new secrets
+                    # backup list of known secrets
+                    cp -pr .secrets.baseline .secrets.new
+                    # find secrets in the repository
+                    detect-secrets scan --disable-plugin AbsolutePathDetectorExperimental --all-files --baseline .secrets.new --exclude-files '\\.secrets..*' --exclude-files '\\.git.*' --exclude-files 'test_results'
+                    # break build when new secrets discovered
+                    # function compares baseline/new secrets w/o listing results -- success(0) when new secret found
+
+                    jq -r '.results | keys[] as $key | "\\($key),\\(.[$key] | .[] | .hashed_secret)"' ".secrets.baseline" | sort > /data/tmp/.secrets.1
+                    jq -r '.results | keys[] as $key | "\\($key),\\(.[$key] | .[] | .hashed_secret)"' ".secrets.new" | sort > /data/tmp/.secrets.2
+
+                    diff /data/tmp/.secrets.1 /data/tmp/.secrets.2 > .secrets.diff || true
+                    '''
+
+                archiveArtifacts artifacts: ".secrets.new", allowEmptyArchive: true, fingerprint: true
+                archiveArtifacts artifacts: ".secrets.diff", allowEmptyArchive: true, fingerprint: true
+
+                sh label: "Compare scan result to baseline",
+                   script: """
+                   #!/bin/bash
```

Comment:
Again, lets get this extracted into a shell script for maintainability

Reply:
...and done!